backend/packages/Upgrade/src/api/services/SegmentService.ts

```diff
@@ -152,46 +152,51 @@ export class SegmentService {
     return await this.segmentRepository.deleteSegment(id, logger);
   }
 
-  public async importSegment(segment: SegmentInputValidator, logger: UpgradeLogger): Promise<Segment> {
-    const duplicateSegment = await this.segmentRepository.findOne(segment.id);
-    if (duplicateSegment && segment.id !== undefined) {
-      const error = new Error('Duplicate segment');
-      (error as any).type = SERVER_ERROR.QUERY_FAILED;
-      logger.error(error);
-      throw error;
-    }
-
-    // check for each subSegment to exists
-    const allSegments = await this.segmentRepository.getAllSegments(logger);
-    segment.subSegmentIds.forEach((subSegmentId) => {
-      const subSegment = allSegments.find((segmentId) => subSegmentId === segmentId.id);
-      if (!subSegment) {
-        const error = new Error(
-          'SubSegment: ' + subSegmentId + ' not found. Please import subSegment and link in experiment.'
-        );
+  public async importSegment(segments: SegmentInputValidator[], logger: UpgradeLogger): Promise<Segment[]> {
+    const allAddedSegments: Segment[] = [];
+    for (const segment of segments) {
+      const duplicateSegment = await this.segmentRepository.findOne(segment.id);
```

Comment:
Find all the segments in a single query.

Reply:
resolved